### PR TITLE
Fix panelapp updating error

### LINKED ISSUE #4776: Key missing error when updating PanelApp green Panel
```
2024-08-22 11:04:23 hasta.scilifelab.se scout.commands.update.panelapp[204759] INFO Updating/creating the PanelApp Green gene panel
2024-08-22 11:04:23 hasta.scilifelab.se scout.load.panel[204759] INFO Fetching all panel app panels
2024-08-22 11:04:24 hasta.scilifelab.se scout.utils.scout_requests[204759] INFO Return in json
2024-08-22 11:04:24 hasta.scilifelab.se scout.commands.update.panelapp[204759] ERROR 'ensembl_id'
Aborted!
```

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4776

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Locally, first load all genes, then ---> `scout --demo update panelapp-green -i cust000 --force`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/load/panel.py [hunk 1]
```diff
@@ -117,27 +118,33 @@ def _panelapp_panel_ids() -> List[str]:
...
+        ensembl_to_hgnc_map: dict[str, int]
+        hgnc_to_ensembl_map: [int, str]
+
 
     Returns:
         parsed_panel(dict). Example:
             {'version': 3.3, 'date': datetime.datetime(2023, 1, 31, 16, 43, 37, 521719), 'display_name': 'Diabetes - neonatal onset - [GREEN]', 'institute': 'cust000', 'panel_type': 'clinical', 'genes': [list of genes], 'panel_id': '55a9041e22c1fc6711b0c6c0'}
 
     """
-    ensembl_gene_hgnc_id_map: Dict[str, int] = adapter.ensembl_to_hgnc_id_mapping()
```
Comment (northwestwitch) [inline scout/load/panel.py:133]: This was really bad, it was fetching both mappings for each PanelApp panel that was parsed :flushed:

**Comment by northwestwitch** [comment]
Local test:

image

File: scout/load/panel.py [hunk 2]
```diff
@@ -117,27 +117,33 @@ def _panelapp_panel_ids() -> List[str]:
...
```
Comment (dnil) [inline scout/load/panel.py:133]: 👍

File: scout/load/panel.py [hunk 3]
```diff
@@ -117,27 +117,33 @@ def _panelapp_panel_ids() -> List[str]:
...
-    hgnc_symbol_ensembl_gene_map: Dict[str, str] = adapter.hgnc_symbol_ensembl_id_mapping()
-
     json_lines = fetch_resource(PANELAPP_BASE_URL.format("get_panel") + panel_id, json=True)
     parsed_panel = parse_panel_app_panel(
         panel_info=json_lines["result"],
-        ensembl_gene_hgnc_id_map=ensembl_gene_hgnc_id_map,
-        hgnc_symbol_ensembl_gene_map=hgnc_symbol_ensembl_gene_map,
+        ensembl_gene_hgnc_id_map=ensembl_id_to_hgnc_id_map,
```
Comment (dnil) [inline scout/load/panel.py:145]: 👍

**Review by dnil — APPROVED** [review]
Looks good, the numbers check out and it is a bug fix!

**Comment by northwestwitch** [comment]
For completeness sake I also tested loading single PanelApp panels. works:

image

**Pull request merged**